### PR TITLE
fix: Make plugin instead of spec

### PR DESCRIPTION
**Pull request by yevgenypats**
#### Summary

I think this should be simpler solution then https://github.com/cloudquery/plugin-sdk/pull/270 changing only the user layer without breaking also all our official and community plugins code-wise.

### CODE REVIEW

File: specs/destination_test.go [hunk 1]
```diff
@@ -109,7 +109,8 @@ var destinationUnmarshalSpecValidateTestCases = []struct {
 	{
 		"required_name",
 		`kind: destination
-spec:`,
+spec:
+  version: v1.1.0`,
```
Comment (disq) [inline specs/destination_test.go:113]: Unused field? Wouldn't this fail because of `DisallowUnknownFields`?

File: specs/spec.go [hunk 2]
```diff
@@ -73,9 +74,21 @@ func (s *Spec) UnmarshalJSON(data []byte) error {
...
+		if err != nil {
+			return err
+		}
+	} else if t.Plugin != nil {
+		b, err = json.Marshal(t.Plugin)
+		if err != nil {
+			return err
+		}
+	} else {
+		return fmt.Errorf("plugin is required and shouldn't be empty")
```
Comment (disq) [inline specs/spec.go:91]: This check wasn't here before, new functionality?

**Review by disq — CHANGES_REQUESTED** [review]
to reiterate: You're suggesting to keep all code complexity (and developer "unfriendliness") associated with double `spec` and not even checking for the new syntax (with `plugin`) in tests?

File: specs/spec.go (hunk 2 above)
Comment (yevgenypats) [reply specs/spec.go:91]: yeah I think it was a bug prob but can go either way

File: specs/destination_test.go (hunk 1 above)
Comment (yevgenypats) [reply specs/destination_test.go:113]: there are no unknown fields. version is known.

**Comment by yevgenypats** [comment]
> to reiterate: You're suggesting to keep all code complexity (and developer "unfriendliness") associated with double `spec` and not even checking for the new syntax (with `plugin`) in tests?

Exactly. I think the value of changing `spec` to `plugin` is not that big in the first place but let's say we change and in a backward compatible way and now we need to update all documentation, I think we shouldn't change now also the codewise, let's just leave as is and avoid updating all plugins for that, I just feel we took the naming thing a bit too far.

**Comment by yevgenypats** [comment]
> > to reiterate: You're suggesting to keep all code complexity (and developer "unfriendliness") associated with double `spec` and not even checking for the new syntax (with `plugin`) in tests?
> 
> Exactly. I think the value of changing `spec` to `plugin` is not that big in the first place but let's say we change and in a backward compatible way and now we need to update all documentation, I think we shouldn't change now also the codewise, let's just leave as is and avoid updating all plugins for that, I just feel we took the naming thing a bit too far.

At this point Im not even sure we really need to change it all given the hussle just for this change and then the docs just for the sake of the name.

File: specs/destination_test.go (hunk 1 above)
Comment (disq) [reply specs/destination_test.go:113]: ah sorry I thought of something like the spec version for some reason

**Comment by disq** [comment]
> At this point Im not even sure we really need to change it all given the hussle just for this change and then the docs just for the sake of the name.

You're still unsure this change is needed that explains why this PR just renames the field from the user standpoint but doesn't change anything else internally.

**Comment by yevgenypats** [comment]
Closing as no-fix for now and will be gathering more experience to see if the `spec` -> `plugin` name is a real issue

**Pull request closed without merging**